### PR TITLE
Amp 595 hmgm runner

### PR DESCRIPTION
**Pull request by dfische3**
Creates a hmgm runner to handle a finite queue of hmgm tasks.  Included is a sample hmgm task (tools/hmgm/hmgm_sample.xml).  The sample code in hmgm_sample_wrapper.py is mostly to illustrate how an hmgm could work.  Right now it randomly exits based on pseudo random numbers. 

The runner is set up to listen to exit codes from the hmgm.  Exit Code 1 = Requeue, Exit Code -1 = Fail, Exit Code 0 = Task complete.

Each tool that needs to use this pool of workers can be setup in /config/job_conf.xml under the tools section.  Currently, each worker sleeps 5 seconds when a job needs to be re-queued.  The current configuration is 4 total workers, 1 of which are dedicated to hmgms.  We can adjust this as necessary.

### CODE REVIEW

File: lib/galaxy/jobs/runners/hmgm.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+            stdout_file.close()
+            stderr_file.close()
+            log.debug('execution finished: %s' % command_line)
+        except Exception:
+            log.exception("failure running job %d", job_wrapper.job_id)
+            self._fail_job_local(job_wrapper, "failure running job")
+            return -1
+        
+        return exit_code
+
```
Comment (yingfeng-iu) [inline lib/galaxy/jobs/runners/hmgm.py]: Just a small suggestion: could you mark the change (like begin change, end change) in the "mostly copied" methods, so if we have an upgrade of Galaxy in the future, we can merge our customization more easily?

File: config/tool_conf.xml [hunk 2]
```diff
@@ -31,6 +31,9 @@
     <tool file="spacy/spacy.xml"/>
     <tool file="aws/aws_comprehend.xml"/>
   </section>
+  <section id="hmgm" name="Hmgm">
+    <tool id="hmgmtest" file="hmgm/hmgm_sample.xml"/>
```
Comment (yingfeng-iu) [inline config/tool_conf.xml]: Could you change name to "Human MGM" I'll ask Maria is she is OK with ID being hmgm. I think it's clear enough to user, but she may want to spell out. Or we can make this change later, as we work on the first HMGM tool.

File: lib/galaxy/jobs/runners/hmgm.py (hunk 1 above)
Comment (dfische3) [reply lib/galaxy/jobs/runners/hmgm.py]: Done

File: config/tool_conf.xml (hunk 2 above)
Comment (dfische3) [reply config/tool_conf.xml]: The section has been renamed to Human MGM.  I don't think the section ID is used anywhere, is it?

Comment (yingfeng-iu) [reply config/tool_conf.xml]: Looks good. Section ID probably doesn't matter.

**Pull request merged**